### ISSUE TITLE
Wrong compiler platform selected with Visual Studio 16 2019 kit

### ISSUE THREAD

**Issue by johannesstricker** [microsoft/vscode-cmake-tools#964]
### Brief Issue Summary
The automatically detected kit for Visual Studio 16 2019 for the x86 platform incorrectly uses the x64 compiler. This can be fixed by adding `"platform": "Win32"` to the kit configuration in `cmake-tools-kits.json` and changing `"visualStudioArchitecture"` from `"x86"` to `"Win32"`.

### Expected:
1. Scan for kits
2. Select the `Visual Studio 16 2019 - x86` kit
3. Configure and build
4. Should generate and build with the x86 compiler.

### Apparent Behavior:
4. The compiler that's being used is the x64 compiler.

### Fix:
Open the `cmake-tools-kits.json` and change
```
  {
    "name": "Visual Studio Enterprise 2019 Release - x86",
    "visualStudio": "11e6a10a",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019"
    }
  }
```
to
```
  {
    "name": "Visual Studio Enterprise 2019 Release - x86",
    "visualStudio": "11e6a10a",
    "visualStudioArchitecture": "Win32",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "Win32"
    }
  }
```

### Platform and Versions
- **Operating System**: Windows 10
- **CMake Version**: 3.16
- **VSCode Version**: 1.40.2
- **CMake Tools Extension Version**: 1.2.3
- **Compiler/Toolchain**: Visual C++ 2019 Enterprise

**Comment by H-G-Hristov** [comment]
Are you sure about this line: `"visualStudioArchitecture": "x86",` -> `"visualStudioArchitecture": "Win32"`,. 
I'll see what I can do later about that issue.

**Comment by Zingam** [comment]
@johannesstricker  Could you, please, provide the Debug Console output.

**Comment by andreeis** [comment]
I am not reproducing. Can you attach a standalone repro?

**Comment by bobbrow** [comment]
@andreeis I don't think it will repro with Ninja, but it should with the Visual Studio generator. @johannesstricker, using Ninja as your generator might also be a workaround.

**Comment by andreeis** [comment]
CMake Tools 1.4.0 Beta is available on GitHub.
https://github.com/microsoft/vscode-cmake-tools/releases/tag/1.4.0-beta

Please try it out and let us know if you encounter any issues.

**Comment by andreeis** [comment]
This fix is available in CMake Tools #1.4.0 which has been released.
